### PR TITLE
add netavark plugin support

### PR DESCRIPTION
**Pull request by Luap99**
Add a plugin system for netavark,
check the plugin-API.md file for a description.

see commits for details

### CODE REVIEW

**Comment by Luap99** [comment]
@b-m-f Sorry I have a lot going on at the moment, I try to get back to this soon. I will talk with the team to see how we can proceed with your PR.

**Comment by Luap99** [comment]
@baude @mheon @flouthoc PTAL

**Comment by mheon** [comment]
Can the WIP be dropped?

File: examples/host-device-plugin.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                        subnets.push(types::NetAddress {
+                            gateway: None,
+                            ipnet: net,
+                        })
+                    }
+                }
+            }
+        }
+
+        //let routes = host.netlink.dump_routes()?;
```
Comment (mheon) [inline examples/host-device-plugin.rs:83]: Can this be dropped?

**Comment by Luap99** [comment]
> Can the WIP be dropped?

I want to make sure this is not merged yet, I still need to test with c/common and podman. Also if we plan to do another release this week this should wait for the next one.
This code here is done I only plan to add some further documentation.

**Comment by b-m-f** [comment]
@Luap99 should I start rebasing thr WireGuard changes onto this branch?

Im also wondering what to do about my PRs in the podman and containers/common repos.
Will they stay relevant when using the plugin system or will there be a new way of passing data down to netavark Plugins?

**Comment by Luap99** [comment]
@b-m-f Please keep your PR rebased on main.

@mheon @flouthoc @baude PTAL, this should be ready from my side.

File: plugin-API.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+The create subcommand creates a new network config for podman.
+The subcommand will receive the JSON network config via STDIN. Podman will populate
+the network name and ID before calling the plugin. The name and ID cannot be changed
+by the plugin. The driver name must also not be changed. All other config
+fields can be changed in the plugin.
+
+Other fields such as subnet and options will also be populated by podman when
+these options are set on the podman network create command, i.e. `--subnet`
+and `--option`. The plugin should validate the given values and error out for
```
Comment (baude) [inline plugin-API.md:19]: if you end up having to address any other comments, consider dropping the use of "should"; i.e. "The plugin validates (or will validate) ... "

**Comment by baude** [comment]
this is great work ... LGTM thanks @Luap99

File: plugin-API.md (hunk 2 above)
Comment (Luap99) [reply plugin-API.md]: fixed

**Pull request merged**